src/modules/add-site/components/blueprints.tsx

```diff
@@ -39,6 +40,114 @@ interface DataViewBlueprint extends Blueprint {
 	categories: string[];
 }
 
+interface BlueprintDetailsModalProps {
+	warnings: Array< { feature: string; reason: string } > | null;
+	validationError: string | null;
+	fileName: string;
+	isOpen: boolean;
+	onClose: () => void;
+}
+
+function BlueprintDetailsModal( {
```

Comment:
This dialog is only about warning/error, so maybe I would propose something more specific as `BlueprintIssuesModal`
```suggestion
function BlueprintIssuesModal( {
```